examples/spark2/src/main/scala/org/apache/carbondata/examples/CompareTest.scala

```diff
@@ -0,0 +1,122 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.examples
+
+import org.apache.spark.sql.{DataFrame, Row, SaveMode, SparkSession}
+
+import org.apache.carbondata.core.util.CarbonProperties
+
+// scalastyle:off println
+object CompareTest {
+
+  val parquetTableName = "comparetest_parquet"
+  val carbonTableName = "comparetest_carbon"
+
+  private def generateDataFrame(spark: SparkSession): DataFrame = {
+    import spark.implicits._
+    spark.sparkContext.parallelize(1 to 10 * 1000 * 1000, 4)
+        .map(x => ("i" + x, "p" + x % 10, "j" + x % 100, x, x + 1, (x + 7) % 21, (x + 5) / 43, x
+            * 5))
+        .toDF("id", "country", "city", "c4", "c5", "c6", "c7", "c8")
+  }
+
+  private def loadParquetTable(spark: SparkSession, input: DataFrame): Long = timeit {
+    input.write.mode(SaveMode.Overwrite).parquet(parquetTableName)
+  }
+
+  private def loadCarbonTable(spark: SparkSession, input: DataFrame): Long = {
+    spark.sql(s"drop table if exists $carbonTableName")
+    timeit {
+      input.write
+          .format("carbondata")
+          .option("tableName", carbonTableName)
+          .option("tempCSV", "false")
+          .option("single_pass", "true")
+          .option("dictionary_exclude", "id") // id is high cardinality column
+          .mode(SaveMode.Overwrite)
+          .save()
+    }
+  }
+
+  private def prepareTable(spark: SparkSession): Unit = {
+    val df = generateDataFrame(spark).cache()
+    println(s"loading dataframe into table, schema: ${df.schema}")
+    val loadParquetTime = loadParquetTable(spark, df)
+    val loadCarbonTime = loadCarbonTable(spark, df)
+    println(s"load completed, time: $loadParquetTime, $loadCarbonTime")
+    spark.read.parquet(parquetTableName).registerTempTable(parquetTableName)
+  }
+
+  private def runQuery(spark: SparkSession): Unit = {
+    val test = Array(
+      "select count(*) from $table",
+      "select sum(c4) from $table",
+      "select sum(c4), sum(c5) from $table",
+      "select sum(c4), sum(c5), sum(c6) from $table",
+      "select sum(c4), sum(c5), sum(c6), sum(c7) from $table",
+      "select sum(c4), sum(c5), sum(c6), sum(c7), avg(c8) from $table",
+      "select * from $table where id = 'i9999999' ",
+      "select * from $table where country = 'p9' ",
+      "select * from $table where city = 'j99' ",
+      "select * from $table where c4 < 1000 "
```

Comment:
please add more testcase, for example:
 "select sum(c4) from $table where id like 'i1%' "
 "select sum(c4) from $table where id like '%10' "
 "select sum(c4) from $table where id like '%xyz%' "